frontend/src/routes.tsx

```diff
@@ -77,16 +77,25 @@ export default function AppRoutes () {
     ]);
 
     useMemo(() => {
-        dispatch(getConfiguration({configScope: 'global'}));
+        let configFetched = false;
+
+        async function fetchAppConfig () {
+            if (!configFetched) {
+                await dispatch(getConfiguration({configScope: 'global'}));
+            }
+            configFetched = true;
+        }
+
+        fetchAppConfig();
```

Comment:
cant this be summarized to 
```suggestion
useMemo(async () => {
      await dispatch(getConfiguration({configScope: 'global'}));
```

Reply:
The [React documentation](https://react.dev/learn/synchronizing-with-effects#fetching-data) makes it a point to include the boolean to keep track of the request has already been made. I could probably get away with:

```
    useMemo(async () => {
        let configFetched = false;
        if (!configFetched) {
            await dispatch(getConfiguration({configScope: 'global'}));
        }
        configFetched = true;
        if (configLoadError) {
            notificationService.generateNotification(
                'Error loading app configuration. Restrictive default policy has been applied in its place. Consult with system admin to resolve issue.',
                'error'
            );
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        configLoadError,
        dispatch,
    ]);
```

Reply:
To really match their example, I'm currently testing:
```
    useMemo(async () => {
        let configFetched = false;
        if (!configFetched) {
            await dispatch(getConfiguration({configScope: 'global'}));
        }
        
        if (configLoadError) {
            notificationService.generateNotification(
                'Error loading app configuration. Restrictive default policy has been applied in its place. Consult with system admin to resolve issue.',
                'error'
            );
        }
        return () => {
            configFetched = true;
        };
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        configLoadError,
        dispatch,
    ]);
```